### PR TITLE
Fix JsonInput micro-batching support

### PR DESCRIPTION
**Pull request by bojiang**
## Description

* Fix the inconsistency of current JsonInput API for handling single prediction request and handling batch request
* Added legacy JSON input class for an easy tranistion

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [x] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/adapters/json_input.py [hunk 1]
```diff
@@ -37,16 +37,15 @@ def __init__(self, is_batch_input=False, **base_kwargs):
...
-        if request.content_type == "application/json":
-            parsed_json = json.loads(request.get_data(as_text=True))
-        else:
+        if request.content_type != "application/json":
             raise BadInput(
                 "Request content-type must be 'application/json' for this "
                 "BentoService API"
             )
-
-        result = func(parsed_json)
```
Comment (bojiang) [inline bentoml/adapters/json_input.py:48]: At least it should be func([parsed_json])

Comment (parano) [reply bentoml/adapters/json_input.py:48]: This will be a regression right? Should we consider something like the `LegacyImageHandler`?

Comment (bojiang) [reply bentoml/adapters/json_input.py:48]: LegacyJsonInput？

File: bentoml/adapters/json_input.py [hunk 2]
```diff
@@ -26,8 +26,30 @@
...
+        ```
+        @bentoml.api(input=LegacyJsonInput())
+        def predict(self, parsed_json):
+            result = do_something_to_json(parsed_json)
+            return result
+        ```
+        
+    --->
+        
+        ```
```
Comment (bojiang) [inline bentoml/adapters/json_input.py:46]: pls help to refine instructions here @parano

**Pull request merged**